modules/openapi-generator/src/main/resources/python/api_test.mustache

```diff
@@ -5,11 +5,23 @@
 from __future__ import absolute_import
 
 import unittest
+import datetime
 
 import {{packageName}}
 from {{apiPackage}}.{{classVarName}} import {{classname}}  # noqa: E501
 from {{packageName}}.rest import ApiException
 
+{{#operations}}
+{{#operation}}
+{{#allParams}}
+{{#required}}
+{{#isModel}}
+import test_{{paramName}}
```

Comment:
why have you added this import?

Reply:
When an API has a required parameter that is an object, I need to instantiate the object to be able to invoke the API.
I import "test_*.py" to access functions that create a default object.

Reply:
Ah, I understand better now.
In your update below it looks like it would produce:
```
test_{{paramName}}.Test{{{dataType}}}.test{{{dataType}}}
--> test_cat.TestCat.testCat
```
If you do it that way, I'm concerned that you are running a test, and not returning a model.
What if in each model test file you had a method make_instance.
That way you could use the import
`from . import test_{{dataType}}`
then when you use it you could run
`{{paramName}} = test_{{dataType}}.make_instance()`

Reply:
Hi!
I split the test on the model into `make_instance()` and the actual test, as you suggested.

I considered to do `from . import Test{{dataType}}`, but the result is:
`ImportError: attempted relative import with no known parent package`
I played around with `-DdebugModels -DdebugOperations`, looking for snake-cased names of test models, since they are used as file names by the code generator.
For the api test, I found them only as paramName. For the model test, there are no snake-case names.

I am now using `from test_{{paramName}} import Test{{dataType}}` for the api tests, and I load model tests dynamically (`imported{{dataType}} = import_module('test_' + camelToSnake("{{dataType}}"))`) for the model tests.

I can imagine two alternatives:
 - work on the java code of openapi-generator, to add snake-cased model names to the data available to the models and operations
 - replicate all the model code I need in all the tests making use of the model. Thus, an api test would have for example the code to instantiate all the models that are used by the api calls.

I would like to know what you think about the approaches.

Reply:
Are you working on solving this problem in the models and operations both at the same time? Maybe it makes sense to solve it in the models first and fix the operations in a later PR.
Looking at the above options, replication is preferred because adding data to the java classes is a pain and we could use a mustache file. I am not sure if replication will work though, in operations we do not have the CodegenModel object, we have a CodegenParameter.

Have you considered adding an __init__.py file to your tests directory? That should solve your import error.

Reply:
Hi, I added . to PYTHONPATH, put all files in the test folder in its `__init__.py`, and took care of the models. I will now get through your other remarks, especially the one regarding using "examples".

Reply:
Thanks for making those changes. As long as the code used to set paramName is the same as the code used to set test_modelname.py then we should be good here

Reply:
Can you resolve this conversation?